pax-manifest/src/utils.rs

```diff
@@ -0,0 +1,130 @@
+use crate::{LiteralBlockDefinition, SettingElement, Token, TokenType, ValueDefinition};
+
+use pest::iterators::Pair;
+use pest::Parser;
+use pest_derive::Parser;
+
+#[derive(Parser)]
+#[grammar = "pax.pest"]
+pub struct PaxParser;
+
+//What do to with location info?
+//a lot of functionality is copied atm
+pub fn to_value_definition(raw_value: &str) -> Option<ValueDefinition> {
+    let mut values = PaxParser::parse(Rule::any_template_value, raw_value).ok()?; //parse using the normal rules
+    if values.as_str() != raw_value {
+        //didn't match entire string -> don't commit
+        return None;
+    }
+    let value = values.next()?.into_inner().next()?;
+    let res = match value.as_rule() {
+        Rule::literal_value => {
+            let literal_value_token =
+                Token::new_only_raw(value.as_str().to_string(), TokenType::LiteralValue);
+            ValueDefinition::LiteralValue(literal_value_token)
+        }
+        Rule::literal_object => {
+            ValueDefinition::Block(derive_value_definition_from_literal_object_pair(value))
+        }
+        Rule::expression_body => {
+            let expression_token =
+                Token::new_only_raw(raw_value.to_string(), TokenType::Expression);
+            ValueDefinition::Expression(expression_token, None)
+        }
+        Rule::identifier => {
+            let identifier_token =
+                Token::new_only_raw(value.as_str().to_string(), TokenType::Identifier);
+            ValueDefinition::Identifier(identifier_token, None)
+        }
+        _ => {
+            unreachable!("Parsing error 3342638857230: {:?}", value.as_rule());
+        }
+    };
+    Some(res)
+}
+
+//--------------------------------------------------------------------------------------------------------
+// Everything below is very similar to functions in pax-compiler, and should be consolidated at some point
+//--------------------------------------------------------------------------------------------------------
+
+fn derive_value_definition_from_literal_object_pair(
```

Comment:
The original compiler function builds token_values as well for each token type, otherwise very similar! Can't think of a very nice way of consolidating them atm.

Reply:
let's use the pax-compiler one here. It doesn't matter if we populate the token_value since it will never be read. It's more important not to duplicate the code. We can clean this up when we get rid of token later on.